### ISSUE TITLE
DescribeConsumerGroups kafka: insufficient data to decode packet, more bytes expected

### ISSUE THREAD

**Issue by CodyGuo** [Shopify/sarama#1717]
##### Versions

*Please specify real version numbers or git SHAs, not just "Latest" since that changes fairly regularly.*

| Sarama | Kafka | Go |
|--------|-------|----|
| v1.26.4 |kafka_2.12-2.3.0  | 1.14.2 |

##### Configuration

What configuration values are you using for Sarama and Kafka?
``` go
dcgs, err := k.cluster.DescribeConsumerGroups(groups)
checkErr(err)
for _, dcg := range dcgs {
    fmt.Printf("state: %s, id: %s\n", dcg.State, dcg.GroupId)
    for name, member := range dcg.Members {
        fmt.Println(name, member.ClientHost, member.ClientId)
        fmt.Printf("%q, %q\n", member.MemberMetadata, member.MemberAssignment)
        metadata, err := member.GetMemberMetadata()
        checkErr(err)
        fmt.Println(metadata.Topics)
    }
}
```

##### Logs

When filing an issue please provide logs from Sarama and Kafka if at all
possible. You can set `sarama.Logger` to a `log.Logger` to capture Sarama debug
output.

<details><summary>logs: CLICK ME</summary>
<p>

```
sarama2020/06/09 11:32:58 client/coordinator coordinator for consumergroup group1 is #2 (10.191.80.143:9092)
state: Empty, id: logstash
state: Stable, id: group1
10.191.80.142-0-cbd3c0b6-b423-43cd-8342-2c190347c78f /10.191.80.142 10.191.80.142-0
"\x00\x00\x00\x00\x00\x02\x00\btomcat-1\x00\x00\x00\f\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00\x03\x00\x00\x00\x04\x00\x00\x00\x05\x00\x00\x00\x06\x00\x00\x00\a\x00\x00\x00\b\x00\x00\x00\t\x00\x00\x00\n\x00\x00\x00\v\x00\ttest-java\x00\x00\x00\f\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00\x03\x00\x00\x00\x04\x00\x00\x00\x05\x00\x00\x00\x06\x00\x00\x00\a\x00\x00\x00\b\x00\x00\x00\t\x00\x00\x00\n\x00\x00\x00\v\x00\x00\x00\x00", "\x00\x00\x00\x00\x00\x02\x00\btomcat-1\x00\x00\x00\f\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00\x03\x00\x00\x00\x04\x00\x00\x00\x05\x00\x00\x00\x06\x00\x00\x00\a\x00\x00\x00\b\x00\x00\x00\t\x00\x00\x00\n\x00\x00\x00\v\x00\ttest-java\x00\x00\x00\f\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00\x03\x00\x00\x00\x04\x00\x00\x00\x05\x00\x00\x00\x06\x00\x00\x00\a\x00\x00\x00\b\x00\x00\x00\t\x00\x00\x00\n\x00\x00\x00\v\x00\x00\x00\x00"
2020/06/09 11:32:58.741255 main.go:91: [FATAL] kafka: insufficient data to decode packet, more bytes expected

```

</p>
</details>

##### Problem Description
kafka: insufficient data to decode packet, more bytes expected

**Comment by dnwe** [comment]
@CodyGuo by coincidence I had stumbled upon this bug recently and investigated a little with @mimaison 

The Kafka Java adminclient only deserializes the member assignments, not the member metadata. 
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L2961-L2962

If you try and get the Java client to deserialize you get the same bug where there’s some unexpected data there so when it reads what it thinks will be a string length it gets a larger value than is possible

> Error reading field 'topics': Error reading string of length 11264, only 335 bytes available

We need to investigate some more to see what the broker is sending

**Comment by dnwe** [comment]
@CodyGuo I think this is actually a server-side defect that was a regression in Kafak 2.3.0, but was fixed in Kafka 2.4.0 under [KAFKA-9150](https://issues.apache.org/jira/browse/KAFKA-9150). If you have control over your Kafka source you can cherry-pick the fix from https://github.com/apache/kafka/pull/7658

**Comment by dnwe** [comment]
I've submitted https://github.com/apache/kafka/pull/8888 to have the fix backported to the 2.3 branch

**Comment by CodyGuo** [comment]
thanks
